lms/djangoapps/instructor/views/api.py

```diff
@@ -392,6 +398,42 @@ def register_and_enroll_students(request, course_id):  # pylint: disable=too-man
             name = student[NAME_INDEX]
             country = student[COUNTRY_INDEX][:2]
 
+            cohort = None  # Optional[CourseUserGroup]
+            course_mode = default_course_mode
+            if len(student) > COHORT_INDEX:
+                # get a cohort name if exists, with caching in `cohorts`
+                cohort_name = student[COHORT_INDEX]
+                if cohort_name and not is_course_cohorted(course_id):
+                    row_errors.append({
+                        'username': username,
+                        'email': email,
+                        'response': _(u'Course is not cohorted but cohort provided. Ignoring cohort.')
+                    })
+                    cohort = None
+                elif cohort_name and cohort_name in cohorts:
+                    cohort = cohorts[cohort_name]
+                elif cohort_name:
+                    # if cohort doesn't exist; ignore
```

Comment:
Nit: reword for clarity. It's not immediately obvious what you mean by "ignore".